sqlglot/schema.py

```diff
@@ -163,12 +163,29 @@ def column_names(self, table: exp.Table | str, only_visible: bool = False) -> t.
         visible = _nested_get(self.visible, *zip(self.supported_table_args, args))  # type: ignore
         return [col for col in columns if col in visible]  # type: ignore
 
-    def get_column_type(self, table: exp.Table | str, column: exp.Column) -> exp.DataType.Type:
+    def get_column_type(
+        self, table: exp.Table | str, column: exp.Column | str
+    ) -> exp.DataType.Type:
         try:
-            schema_type = self.schema.get(table.name, {}).get(column.name).upper()  # type: ignore
+            if isinstance(table, exp.Table):
+                table_args = [
+                    table.text(p)
+                    for p in self.supported_table_args or self._get_table_args_from_table(table)
+                ]
+                table_schema = _nested_get(
+                    self.schema,
+                    *zip(self.supported_table_args, table_args),
```

Comment:
line 173 assumes table args can be empty, but then you use it here

perhaps better to do 

supported_table_args = self.supported_table_args or self._get

then you can use this throughout

Reply:
Done